fix: ensure GetCustomMarkupExtensionValue checks for nested extensions.

beto-rodriguez opened the pull request:
Ensures GetCustomMarkupExtensionValue is recursive and evaluates for netsed custom Xaml extensions.

**GitHub Issue:** closes #21490

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type:

<!--
Copy the labels that apply to this PR and paste them above:

- 🐞 Bugfix
- ✨ Feature
- 🎨 Code style update (formatting)
- 🔄 Refactoring (no functional changes, no api changes)
- 🏗️ Build or CI related changes
- 📚 Documentation content changes
- 🤖 Project automation
- 💬 Other... (Please describe)

-->

- 🐞 Bugfix

## What is the current behavior? 🤔

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

When using custom markup nested extensions, the XAML source generator is not designed to parse properties that are again a custom markup extension, so the next Xaml fails:

```xml
<lvc:CartesianChart>
	<lvc:CartesianChart.Series>
		<lvc:SeriesCollection>
			<lvc:XamlLineSeries
				SeriesName="Mary"
				Values="{lvc:Values FromString='5, 10, 8, 4'}"
				Stroke="{lvc:SolidColorPaint
					Color='#6495ED', 
					StrokeCap=Round, 
					StrokeWidth=10, 
					PathEffect={lvc:Dashed Array='30, 20'}}"/>
		</lvc:SeriesCollection>
	</lvc:CartesianChart.Series>
</lvc:CartesianChart>
```

It fails because the custom `SolidColorPaintExtension` sets the `PathEffect` property using the `Dashed` custom extension, this is currently not supported.

## What is the new behavior? 🚀

<!-- Please describe the new behavior after your modifications. -->

This PR ensures that property members used in a custom markup extension evaluate whether the property is a custom markup extension, if it is then it recursively calls `GetCustomMarkupExtensionValue` to resolve the correct type and value for the property.

## PR Checklist ✅

Please check if your PR fulfills the following requirements:

- [x] 📝 Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [x] 🧪 Added [Runtime tests, UI tests, or a manual test sample](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] 📚 Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] 🖼️ Validated PR `Screenshots Compare Test Run` results.
- [x] ❗ Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

No breaking changes.

## Other information ℹ️

<!-- Please provide any additional information if necessary -->

beto-rodriguez committed:
fix: ensure GetCustomMarkupExtensionValue checks for nested extensions.

Ensures GetCustomMarkupExtensionValue is recursive and evaluates for netsed custom Xaml extensions.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/unoplatform/uno?pullRequest=21493) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/unoplatform/uno?pullRequest=21493) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/unoplatform/uno?pullRequest=21493) it.</sub>

beto-rodriguez commented:
> ⚠️⚠️ The build [176496](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=176496) has failed on Uno.UI - CI.

Hello, I'm confused, it seems that those error logs are unrelated to my changes, I just changed the source generator but fails are related to iOS and Android natives.

Then there are another kind of errors:

> When_TouchScrollDownWithInertia_Then_DoNotChainInertia()

But I checked that test and it is not even using Xaml generated controls.

I hope the errors are not in my changes, but feedback is appreciated in case I need to improve this PR.

MartinZikmund commented:
@beto-rodriguez the failing test was just flaky, only thing needed is a runtime test to validate the change

beto-rodriguez commented:
@MartinZikmund Thanks! I'll add that test asap!

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21493/docs/index.html

beto-rodriguez committed:
test: add new MultiLevelNesting test

ensures custom markup extensions work when nested

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21493/wasm-skia-net9/index.html

beto-rodriguez committed:
fix: remove SG double bracketDepth check

this double check seems unecessary. It also prevent the source generator to parse properly multi/nested xaml extensions.

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21493/docs/index.html

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21493/wasm-skia-net9/index.html
